onnxruntime/core/providers/cpu/reduction/reduction_ops.cc

```diff
@@ -490,13 +492,10 @@ Status ReduceMean<T>::Compute(OpKernelContext* ctx) const {
 
   if (no_transpose) {
     const T* input_data = ctx->Input<Tensor>(0)->template Data<T>();
-
-#ifdef _OPENMP
-#pragma omp parallel for
-#endif
-    for (int64_t i = 0; i < block_size; ++i) {
+    auto lambda = [input_data, blocks, block_size, output_data](ptrdiff_t i) {
       output_data[i] = ConstEigenVectorMap<T>(input_data + (i * blocks), blocks).mean();
```

Comment:
Existing issue in many places, but should there be some heuristic about the amount of work to do before we use the threadpool? Or we need to address that separately on a more global basis?

Reply:
Yeah, I guess we could do it in one of 2 ways: (a) compare block_size (which is the unit of parallelism here just like batch_size) with some threshold and only then use the threadpool (b) see if total_work/num_threads > some threshold and only then use the threadpool. However, I'm not sure sure how do we find such a heuristic? Based on a couple of models? And then how do we make it configurable for tuning?

Reply:
I prefer to use block_size/num_thread as a threshold.

Reply:
Without having a consistent way to tune this threshold through some config, I'm wary to hardcode this threshold here. This applies to not just this op, but many other ops. May be we need a more holistic approach?

Reply:
But that's how mlas works today. If there is a better solution, we may discuss it with Tracy.

Reply:
I checked a bunch of production models where these ops were used. The block_size was maximum 4 from what I saw. I've asked @zhanghuanrong for the model for which he added the 'pragma omp' lines. I don't want to introduce a random heuristic or else it might affect an existing production model.